packages/block-library/src/latest-posts/style.scss

```diff
@@ -16,6 +16,7 @@
 
 		li {
 			clear: both;
+			overflow-wrap: break-word;
```

Comment:
In rare cases, the title also needs to wrap when the "list" option is used, so I think that placing the CSS here instead of limiting it to the grid option makes sense, good catch.